src/tempor/plugins/time_to_event/plugin_ts_xgb.py

```diff
@@ -0,0 +1,242 @@
+            time_bins = surv.columns
+            for t, eval_time in enumerate(time_horizons):
+                nearest = self._find_nearest(time_bins, eval_time)
+                local_preds_[:, t] = np.asarray(1 - surv[nearest])
+            preds_.append(local_preds_)
+        return pd.DataFrame(np.concatenate(preds_, axis=0), columns=time_horizons, index=X.index)
+
+
+# TODO: Docstring.
+@plugins.register_plugin(name="ts_xgb", category="time_to_event")
+class XGBTimeToEventAnalysis(BaseTimeToEventAnalysis):
+    ParamsDefinition = XGBTimeToEventAnalysisParams
+    params: XGBTimeToEventAnalysisParams  # type: ignore
+
+    def __init__(self, **params) -> None:  # pylint: disable=useless-super-delegation
+        """XGB survival analysis model.
+
+        Args:
+            **params:
+                Parameters and defaults as defined in :class:`XGBTimeToEventAnalysisParams`.
+        """
+        super().__init__(**params)
+
+        output_model = XGBSurvivalAnalysis(
+            n_estimators=self.params.xgb_n_estimators,
+            colsample_bynode=self.params.xgb_colsample_bynode,
+            max_depth=self.params.xgb_max_depth,
+            subsample=self.params.xgb_subsample,
+            learning_rate=self.params.xgb_learning_rate,
+            min_child_weight=self.params.xgb_min_child_weight,
+            tree_method=self.params.xgb_tree_method,
+            booster=self.params.xgb_booster,
+            random_state=self.params.random_state,
+            objective=self.params.xgb_objective,
+            strategy=self.params.xgb_strategy,
+            bce_n_iter=self.params.xgb_bce_n_iter,
+            time_points=self.params.xgb_time_points,
+            device=self.params.device,
+        )
+        self.model = EmbTimeToEventAnalysis(
+            output_model=output_model,
+            split=self.params.split,
+            n_layers_hidden=self.params.n_layers_hidden,
+            n_units_hidden=self.params.n_units_hidden,
+            rnn_mode=self.params.rnn_mode,
+            alpha=self.params.alpha,
+            beta=self.params.beta,
+            sigma=self.params.sigma,
+            dropout=self.params.dropout,
+            patience=self.params.patience,
+            lr=self.params.lr,
+            batch_size=self.params.batch_size,
+            n_iter=self.params.n_iter,
+            output_mode=self.params.output_mode,
+            device=self.params.device,
+        )
+
+    def _fit(
+        self,
+        data: dataset.Dataset,
+        *args,
+        **kwargs,
+    ) -> Self:
+        self.model.fit(data, *args, **kwargs)
+        return self
+
+    def _predict(  # type: ignore[override]
+        self,
+        data: dataset.Dataset,
+        horizons: data_typing.TimeIndex,
+        *args,
+        **kwargs,
+    ) -> samples.TimeSeriesSamples:
+        return self.model.predict(data, horizons, *args, **kwargs)
+
+    @staticmethod
+    def hyperparameter_space(*args, **kwargs):
```

Comment:
@bcebere
Looks like some of these hyperparameters are unused / don't match the __init__?
* `xgb_reg_lambda`
* `xgb_reg_alpha`
* `xgb_colsample_bytree`
* `xgb_colsample_bylevel`